subsys/nrf_security/src/drivers/Kconfig

```diff
@@ -4,56 +4,229 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_SUPPORT
+# Legacy config that needs to be removed later. The Oberon PSA driver is now the fallback option
+# for all the crypto operations when hardware acceleration is not available.
+config PSA_CRYPTO_DRIVER_OBERON
+	# The Oberon driver is required to provide HKDF needed for protected storage and bultin keys
+	prompt "Oberon PSA driver" if !(TFM_PARTITION_PROTECTED_STORAGE || TFM_CRYPTO_BUILTIN_KEYS)
 	bool
 	default y
-	depends on MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_KEY_PAIR || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_PUBLIC_KEY || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_AES || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_CHACHA20
+	help
+	  This configuration enables the usage of the Oberon PSA driver. Note that the Oberon
+	  PSA core is always included in the build.
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR
+config PSA_CRYPTO_DRIVER_CC3XX
+	prompt "CryptoCell PSA driver"
+	bool
+	help
+	  This configuration enables the usage of CryptoCell for the supported operations.
+	  Disabling this option will result in all crypto operations being handled by
+	  the Oberon PSA driver. However, please note that disabling this option does not
+	  affect the DRBG algorithms; even when disabled, DRBG will still be provided by CryptoCell.
+	  The CryptoCell PSA driver does not support key size configurations. When enabled,
+	  all supported key sizes are included in the build.
+
+menu "Choose DRBG algorithm (choose only one)"
+config PSA_WANT_ALG_CTR_DRBG
+	prompt "Enable CTR_DRBG"
+	bool
+	default y if !PSA_WANT_ALG_HMAC_DRBG
+	depends on PSA_WANT_GENERATE_RANDOM
+
+config PSA_WANT_ALG_HMAC_DRBG
+	prompt "Enable HMAC_DRBG"
+	bool
+	depends on PSA_WANT_GENERATE_RANDOM
+endmenu
+
+# The Oberon PSA core requires the USE symbols to choose DRBG algorithm.
+config PSA_USE_CTR_DRBG_DRIVER
```

Comment:
PSA_USE_CTR_DRBG_DRIVER and PSA_USE_HMAC_DRBG_DRIVER can be grouped in "Choose DRBG algorithm (choose only one)" menu

Reply:
You are right, I will do that.